src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -26,24 +24,27 @@ public sealed partial class Pipe
         private static readonly SendOrPostCallback s_syncContextExecuteWithoutExecutionContextCallback = ExecuteWithoutExecutionContext!;
         private static readonly Action<object?> s_scheduleWithExecutionContextCallback = ExecuteWithExecutionContext!;
 
-        // This sync objects protects the shared state between the writer and reader (most of this class)
-        private readonly object _sync = new object();
-
-        private readonly MemoryPool<byte>? _pool;
-        private readonly int _minimumSegmentSize;
-        private readonly long _pauseWriterThreshold;
-        private readonly long _resumeWriterThreshold;
-
-        private readonly PipeScheduler _readerScheduler;
-        private readonly PipeScheduler _writerScheduler;
-
         // Mutable struct! Don't make this readonly
         private BufferSegmentStack _bufferSegmentPool;
 
         private readonly DefaultPipeReader _reader;
         private readonly DefaultPipeWriter _writer;
 
-        private readonly bool _useSynchronizationContext;
+        // The options instance
+        private readonly PipeOptions _options;
+        private readonly object _sync = new object();
```

Comment:
I'm in no way saying it's worth it, but _if_ you really wanted to avoid this `new object()` and you wanted to use that buffer segment array you tried to in a previous commit, you could instead still keep this field and just store that array into this field in the ctor.  Then you're reusing the same array instance even the buffer stack is resized.  There are downsides, though, namely you'd be keeping alive what I assume is a much larger object, and if that array referenced other stuff, anything it referenced.